code/game/atoms.dm

```diff
@@ -798,3 +802,13 @@ Proc for attack log creation, because really why not
 	hud_list = new
 	for(var/hud in hud_possible) //Providing huds.
 		hud_list[hud] = image('icons/mob/hud.dmi', src, "")
+
+// For determining if a mover is incorporeal and thus can move through most solid obstacles.
+/atom/proc/incorporeal_movement_check(atom/movable/mover)
+
+	if(ismob(mover))
+		var/mob/M = mover
+		if(M.status_flags & INCORPOREAL)
+			return TRUE
+
+	return FALSE
```

Comment:
This should be an element and use signals

Reply:
This proc should not exist

Reply:
@TiviPlus

Reply:
@Surrealaser This proc still should not exist